invenio_accounts/forms.py

```diff
@@ -44,6 +45,16 @@ def __init__(self, *args, **kwargs):
             if not self.next.data:
                 self.next.data = request.args.get("next", "")
 
+        def validate(self, extra_validators=None):
```

Comment:
During registration we check the domain. We only check when the registration forms require email confirmation.